list-your-project/other-dashboards/fees-adapters.md

```diff
@@ -0,0 +1,308 @@
+
+### 5. Contract Calls
+
+For protocols where fee data is accessible through view functions or requires multiple contract interactions.
+
+```typescript
+import { Adapter, FetchOptions } from "../../adapters/types";
+
+// Contract addresses
+const VOTER = "0xd7ea36ECA1cA3E73bC262A6D05DB01E60AE4AD47";
+const BERO = "0x7838CEc5B11298Ff6a9513Fa385621B765C74174";
+const HONEY = "0xFCBD14DC51f0A4d49d5E53C2E0950e0bC26d0Dce";
+
+// Fee constants 
+const SWAP_FEE = 30n;
+const BORROW_FEE = 250n;
+const DIVISOR = 10000n;
+
+async function fetch(options: FetchOptions) {
+  const dailyFees = options.createBalances();
+  
+  // Get plugin data through contract calls
+  const plugins = await options.api.call({
+    target: VOTER,
+    abi: "address[]:getPlugins",
+  });
+  
+  // Use multiCall for efficiency when making multiple similar calls
+  const bribes = await options.api.multiCall({
+    abi: "function getBribe() returns (address)",
+    calls: plugins.map((plugin) => ({ target: plugin })),
+  });
+  
+  // Collect fee data from events
+  for (const bribe of bribes) {
+    const logs = await options.getLogs({
+      target: bribe,
+      fromBlock: await options.getFromBlock(),
+      toBlock: await options.getToBlock(),
+      eventAbi: "event Bribe__RewardNotified(address indexed rewardToken, uint256 reward)",
+    });
+    
+    logs.forEach((log) => {
+      dailyFees.add(log.rewardToken, log.reward);
+    });
+  }
+  
+  return { dailyFees };
+}
+```
+
+**Example adapters using contract calls**:
+- [Beradrome](https://github.com/DefiLlama/dimension-adapters/blob/master/fees/beradrome/index.ts) - Combines contract calls with event logs to track multiple fee sources
+
+
+### 6. Protocol API Endpoints
+
+Many protocols offer dedicated API endpoints that provide fee and revenue metrics. These should be used with proper validation to ensure accurate reporting.
+
+```typescript
+import fetchURL from "../utils/fetchURL";
+
+const thalaDappURL = "https://app.thala.fi";
+const feesQueryURL = `${thalaDappURL}/api/defillama/trading-fee-chart?timeframe=`;
+
+const fetch = async (timestamp: number) => {
+  const dayFeesQuery = (await fetchURL(feesQueryURL + "1D&endTimestamp=" + timestamp))?.data;
+  const dailyFees = dayFeesQuery.reduce(
+    (sum: number, a: any) => sum + a.value,
+    0
+  );
+  
+  return {
+    dailyFees,
+    dailyRevenue: dailyFees * 0.2, // If protocol keeps 20% of fees
+    timestamp,
+  };
+};
+```
+
+**Example adapter using API endpoints**:
+- [Thalaswap](https://github.com/DefiLlama/dimension-adapters/blob/master/fees/thalaswap.ts) - Fetches metrics from protocol API
+
+### 7. Helper Wrappers for DEXs and Lending Protocols
+
+For common protocol types, DeFiLlama provides specialized helpers that simplify adapter creation:
+
+**Example adapters using helper wrappers**:
+- [Katana](https://github.com/DefiLlama/dimension-adapters/blob/master/fees/katana.ts) - Uses getDexChainFees
```

Comment:
a user reading this won't know what getGraphDimensions2 means, imo better to list examples for forked protocols